File: answers/leetcode/Two Sum/Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[] {0, 0};
        Map<Integer, Integer> numMap = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (numMap.containsKey(diff)) {
                result[0] = numMap.get(diff);
                result[1] = i;
                return result;
            }
            numMap.put(nums[i], i);
        }
        return result;
    }
}

File: answers/leetcode/Two Sum/Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        dict = {}
        for i in xrange(len(nums)):
            x = nums[i]
            if target - x in dict:
                return (dict[target - x] + 1, i + 1)
            dict[x] = i
